java/ql/lib/semmle/code/java/Modifier.qll

```diff
@@ -25,7 +31,15 @@ abstract class Modifiable extends Element {
    * abstract, so `isAbstract()` will hold for them even if `hasModifier("abstract")`
    * does not.
    */
-  predicate hasModifier(string m) { modifiers(this.getAModifier(), m) }
+  predicate hasModifier(string m) {
+    exists(Modifier mod | mod = this.getAModifier() |
+      modifiers(mod, m) and
+      // Kotlin "internal" elements may also get "public" modifiers, so we want to filter those out
+      not exists(Modifier mod2 |
+        hasModifier(this, mod2) and modifiers(mod, "public") and modifiers(mod2, "internal")
+      )
+    )
+  }
```

Comment:
I don't think that we need to change this. `this.getAModifier()` calls `Modifier.getElement()` which has already done the filtering in your change above.

Reply:
Thanks for spotting this. I removed it.

Reply:
Thanks for spotting this. I removed it.